wasm.rkt

```diff
@@ -196,6 +304,14 @@
         ((v_rest ... e ...) L s)
         return-implicit]
 
+   ;;;;;;; FUNCTION CALL FROM TABLE
+   
+   ;; Call Indirect
+   [--> ((v_rest ... v_1 (call-indirect) e ...) L s)
+        ((v_rest ... (fetch-from-table s v_1))(mt-labels mt-locals (e ...)  L) s)
+        call-indirect]
```

Comment:
what if we get rid of fetch-from-table:
`  [--> ((v_rest ... v_1 (call-indirect) e ...) L (s-func s-table s-mem)
        ((v_rest ... (search-table (s-func s-table s-mem) s-table v_1))(mt-labels mt-locals (e ...)  L) s)
        call-indirect]`

Reply:
I have taken a look at the code critically, am i think i need to retain the fetch-from-table. The because, it is the point of entry from the reduction rule that passes the S to it, then it need to deconstruct s into (s-func s-table s-mem) so that it can extract s-table and pass it to search-table which recursively iterates through the s-table to look for the desired function. Thus, removing fetch-from-table won't enable the deconstruction and it technically cumbersome to begin to iterate through S with out a helper function that iterates through the s-table inside S

Reply:
Ok I get it. I think it would be better design if you used Racket "local definitions" to create search-table inside of fetch-from-table, but either way works.